File: JavaScript/2423.js
// Problema: 2423 - Receita de Bolo  | Resposta: Accepted
// Linguagem: JavaScript 12.18 [+2s] | Tempo: 0.085s

var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var [x, o, c] = input.split(' ').map(Number);
var bolos = [Math.floor(x/2), Math.floor(o/3), Math.floor(c/5)];
console.log(Math.min(...bolos));

File: Python/2423.py
# Problema: 2423 - Receita de Bolo | Resposta: Accepted
# Linguagem: Python 3.11 [+1s]     | Tempo: 0.000s

x, o, c = map(int, input().split())
bolos = [x//2, o//3, c//5]
print(min(bolos))